test/provisioner/provision.go

```diff
@@ -121,6 +222,177 @@ func GetInstallOverlay(provider string, installDir string) (InstallOverlay, erro
+	serviceList := &corev1.ServiceList{}
+	err = resources.List(context.TODO(), serviceList)
+	if err != nil {
+		err = fmt.Errorf("Error listing services: %w\n", err)
+		log.Errorf("%v", err)
+		return "", err
+	}
+
+	var matchingService *corev1.Service
+	for i := range serviceList.Items {
+		service := &serviceList.Items[i]
+		if service.Name == deploymentName {
+			matchingService = service
+			break
+		}
+	}
+
+	if matchingService == nil {
+		return "", fmt.Errorf("No service with label selector found")
+	}
+
+	fmt.Printf("Service IP: %s\n", matchingService.Spec.ClusterIP)
+	return matchingService.Spec.ClusterIP, nil
+}
+
+func (p *KeyBrokerService) Deploy(ctx context.Context, cfg *envconf.Config, props map[string]string) error {
+	log.Info("Customize the overlay yaml file")
+	if err := p.installOverlay.Edit(ctx, cfg, props); err != nil {
+		return err
+	}
+
+	originalDir, err := os.Getwd()
+	if err != nil {
+		err = fmt.Errorf("Error getting the current working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	newDirectory := "kbs/kbs/config/kubernetes"
+	err = os.Chdir(newDirectory)
+	if err != nil {
+		err = fmt.Errorf("Error changing the working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// Replace this to use install overlay
+	cmd := exec.Command("kubectl", "apply", "-k", "overlays")
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG="+cfg.KubeconfigFile()))
+	stdoutStderr, err := cmd.CombinedOutput()
+	log.Tracef("%v, output: %s", cmd, stdoutStderr)
+	if err != nil {
+		return err
+	}
+
+	/*
+		log.Info("Install Kbs")
+		if err := p.installOverlay.Apply(ctx, cfg); err != nil {
+			return err
+		}
+	*/
+
+	// Return to the original working directory.
+	err = os.Chdir(originalDir)
+	if err != nil {
+		err = fmt.Errorf("Error changing back to the original working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (p *KeyBrokerService) Delete(ctx context.Context, cfg *envconf.Config) error {
+
+	originalDir, err := os.Getwd()
+	if err != nil {
+		err = fmt.Errorf("Error getting the current working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// Remove kbs deployment
+	newDirectory := "kbs/kbs/config/kubernetes"
+	err = os.Chdir(newDirectory)
+	if err != nil {
+		err = fmt.Errorf("Error changing the working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	log.Info("Delete Kbs deployment")
+	if err := p.installOverlay.Delete(ctx, cfg); err != nil {
+		return err
+	}
+
+	// Return to the original working directory.
+	err = os.Chdir(originalDir)
+	if err != nil {
+		err = fmt.Errorf("Error changing back to the original working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// remove kbs repo
+	directoryPath := "kbs"
+
+	err = os.RemoveAll(directoryPath)
+	if err != nil {
+		err = fmt.Errorf("Error deleting directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	return nil
```

Comment:
i'd really avoid it, unless we're 100% sure it's in a temporary folder. people might symlink their kbs folder and this would erase their unpushed work.